src/general/VersionUtils.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Globalization;
+using Godot;
+
+/// <summary>
+///   Helpers for dealing with Thrive's version number
+/// </summary>
+public static class VersionUtils
+{
+    public static readonly string[] Suffixes =
+    {
+        "PRE-ALPHA",
+        "ALPHA",
+        "BETA",
+        "RC1",
+        "RC2",
+        "RC3",
+        "RC4",
+        "RC5",
+        "RC6",
+        "RC7",
+        "RC8",
+        "RC9",
+    };
+
+    /// <summary>
+    ///   Compare the given version numbers.
+    /// </summary>
+    /// <returns>
+    ///   0 if the versions are the same,
+    ///   a negative integer if a is a smaller (older) version than b and
+    ///   a positive integer if a is a bigger (newer) version than b
+    /// </returns>
+    /// <param name="a">The first version to compare.</param>
+    /// <param name="b">The second version to compare.</param>
+    public static int Compare(string a, string b)
+    {
+        if (a == b)
+        {
+            return 0;
+        }
+
+        char[] separator = { '-' };
+        var aSplit = a.Split(separator, 2);
+        var bSplit = b.Split(separator, 2);
+
+        // Compare the numeric versions
+        try
+        {
+            int versionDiff = Version.Parse(aSplit[0]).CompareTo(Version.Parse(bSplit[0]));
+            if (versionDiff != 0)
+            {
+                return versionDiff;
+            }
+        }
+        catch (Exception e)
+        {
+            GD.PrintErr(e);
+        }
```

Comment:
add a return 0 here, and make the error message also print that version comparison failed, and also print `a` and `b`

Reply:
Would a positive integer be better so the user sees "from newer version, incompatible"?

Reply:
Maybe? I was thinking that situations like sorting versions and such would benefit from considering incomparable versions to be equal.
Or maybe the catch block could be in the SaveListItem as there showing the newer version popup would make sense. And other places can catch the exception only if they need to.

Reply:
Problems should only arise from unparseable versions which either means the save is broken or from a distant future where the versioning scheme has changed.